src/s3path.jl

```diff
@@ -86,15 +86,33 @@ function S3Path(
 end
 
 # To avoid a breaking change.
-function S3Path(str::AbstractString; config::AbstractAWSConfig=global_aws_config(),
-                version::AbstractS3Version=nothing)
+function S3Path(
```

Comment:
Cleaned up this function a little as being able to write `S3Path(str; isdirectory=true)` was handy for debugging the performance issues this PR was trying to solve. If folks want I could make this a separate PR?